cmd/deploy.go

```diff
@@ -216,6 +218,27 @@ var deployCmd = &cobra.Command{
 
 		// print table summary
 		printContainerInspect(c, containers, c.Config.Mgmt.Network, format)
+
+		// exec cmds specified for containers
+		execJSONResult := make(map[string]map[string]map[string]interface{})
+		for _, cont := range containers {
+			name := cont.Labels[clab.NodeNameLabel]
+			if node, ok := c.Nodes[name]; ok && (len(node.Config().Exec) > 0) {
+				rt := node.GetRuntime()
+				contName := strings.TrimLeft(cont.Names[0], "/")
+				if execJSONResult[contName], err = execCmds(ctx, cont, rt, node.Config().Exec, format); err != nil {
+					log.Errorf("Failed to exec cmds for node %s", name)
+				}
+			}
+		}
+		if format == "json" && (len(execJSONResult) > 0) {
+			result, err := json.Marshal(execJSONResult)
+			if err != nil {
+				log.Errorf("Issue converting exec results to json %v", err)
+			}
+			fmt.Println(string(result))
+		}
+
```

Comment:
every node interface has a `PostDeploy()` method. In this method we run stuff that should happen after the node has been deployed

Maybe it makes sense to call the `execCmds` in that postDeploy phase? In that case all the runtime will be initialised already.

Reply:
Yep, i also thought about it. Problem with it is that it looks like it causing unnecessary repetition. Since `execCmds` code, probably will be the same for all types of nodes.

Reply:
okay, that is fair
also we might soon have another action that is intended to be run right after the lab is deployed so this block can grow a bit